core/corehttp/gateway_handler.go

```diff
@@ -16,37 +17,50 @@ import (
 	"github.com/TRON-US/go-mfs"
 	coreiface "github.com/TRON-US/interface-go-btfs-core"
 	ipath "github.com/TRON-US/interface-go-btfs-core/path"
+	"github.com/Workiva/go-datastructures/cache"
 	"github.com/dustin/go-humanize"
 	"github.com/ipfs/go-cid"
 	dag "github.com/ipfs/go-merkledag"
-	"github.com/ipfs/go-path"
+	ipfspath "github.com/ipfs/go-path"
 	"github.com/ipfs/go-path/resolver"
 	routing "github.com/libp2p/go-libp2p-core/routing"
 	"github.com/multiformats/go-multibase"
 )
 
 const (
-	ipfsPathPrefix = "/btfs/"
-	ipnsPathPrefix = "/btns/"
+	ipfsPathPrefix                           = "/btfs/"
+	ipnsPathPrefix                           = "/btns/"
+	GatewayReedSolomonDirectoryCacheCapacity = 10
 )
 
 // gatewayHandler is a HTTP handler that serves BTFS objects (accessible by default at /btfs/<path>)
 // (it serves requests like GET /btfs/QmVRzPKPzNtSrEzBFm2UZfxmPAgnaLke4DMcerbsGGSaFe/link)
 type gatewayHandler struct {
 	config GatewayConfig
 	api    coreiface.CoreAPI
+	rsDirs cache.Cache
 }
 
-func newGatewayHandler(c GatewayConfig, api coreiface.CoreAPI) *gatewayHandler {
+type ReedSolomonDirectory struct {
+	rootPath ipath.Resolved
+	rootDir  files.Directory
+}
+
+func (d ReedSolomonDirectory) Size() uint64 {
+	return uint64(1)
```

Comment:
why is it fixed at 1? 1 means one folder?

Reply:
// Returns one since this Size() is used by the cache.Cache
	// to indicate one directory entry in the cache.